### PR TITLE
Stat tracker methods

### PR DESCRIPTION
**Pull request by IamNorma**
require relative files in stat tracker class and added league stats class methods and tests: 

- best_offense
- worst_offense
- highest_scoring_visitor
- lowest_scoring_visitor
- highest_scoring_home_team
- lowest_scoring_home_team

### CODE REVIEW

File: test/stat_tracker_test.rb [hunk 1]
```diff
@@ -161,4 +161,28 @@ def test_it_can_get_team_with_fewest_tackles
...
+    assert_equal "Utah Royals FC", @stat_tracker.worst_offense
+  end
+
+  def test_it_can_return_highest_scoring_visitor
+    assert_equal "Reign FC", @stat_tracker.highest_scoring_visitor
+  end
+
+  def test_it_can_return_lowest_scoring_visitor
+    assert_equal "Utah Royals FC", @stat_tracker.lowest_scoring_visitor
+  end
```
Comment (dborski) [inline test/stat_tracker_test.rb:179]: The spec harness has the highest scoring visitor as "FC Dallas" and lowest scoring visitor as "San Jose Earthquakes". We may want to double check these methods.

Comment (kiverso) [reply test/stat_tracker_test.rb:179]: agreed, we can get on a call this week and dive into these methods to see if we can figure out where things are getting off.

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -1,3 +1,10 @@
+require_relative 'game'
+require_relative 'team'
+require_relative 'game_team'
+require_relative 'season_stats'
+require_relative 'league_stats'
+require_relative 'team_stats'
```
Comment (kiverso) [inline lib/stat_tracker.rb:6]: great! I kept getting errors running the spec harness and I think this fixes it. nice catch.

File: test/stat_tracker_test.rb (hunk 1 above)
Comment (kiverso) [reply test/stat_tracker_test.rb:179]: Also noticing these tests take a very long time to run, we will need to look into how to make these methods more efficient.

**Pull request merged**